tests/v2/validation/clustertemplates/rbac/clustertemplate_test.go

```diff
@@ -0,0 +1,150 @@
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, ct.provisioningConfig)
+
+	client, err := rancher.NewClient("", testSession) //ct.client
+	require.NoError(ct.T(), err)
+
+	ct.client = client
+
+	if ct.provisioningConfig.RKE1KubernetesVersions == nil {
+		rke1Versions, err := kubernetesversions.ListRKE1AllVersions(ct.client)
+		require.NoError(ct.T(), err)
+
+		ct.provisioningConfig.RKE1KubernetesVersions = []string{rke1Versions[len(rke1Versions)-1]}
+	}
+
+	if ct.provisioningConfig.CNIs == nil {
+		ct.provisioningConfig.CNIs = []string{clustertemplates.CniCalico}
+	}
+
+}
+
+func (ct *ClusterTemplateRKE1RBACTestSuite) TestClusterTemplateEnforcementForStandardUser() {
+	log.Info("Enforcing cluster template while provisioning rke1 clusters")
+
+	clusterEnforcement, err := ct.client.Steve.SteveType(settings.ManagementSetting).ByID(clustertemplates.ClusterEnforcementSetting)
+	require.NoError(ct.T(), err)
+	_, err = settings.UpdateGlobalSettings(ct.client.Steve, clusterEnforcement, clustertemplates.EnabledClusterEnforcementSetting)
+	require.NoError(ct.T(), err)
+
+	verifySetting, err := ct.client.Steve.SteveType(settings.ManagementSetting).ByID(clustertemplates.ClusterEnforcementSetting)
+	require.NoError(ct.T(), err)
+	require.Equal(ct.T(), verifySetting.JSONResp["value"], clustertemplates.EnabledClusterEnforcementSetting)
+
+	log.Info("Creating a cluster template and verifying the user is not able to create the cluster without the required permissions.")
+	clusterTemplate, err := clustertemplates.CreateRkeTemplate(ct.client, nil)
+	require.NoError(ct.T(), err)
+
+	templateRevisonConfig := mgmt.ClusterTemplateRevision{
+		ClusterConfig: &mgmt.ClusterSpecBase{RancherKubernetesEngineConfig: &mgmt.RancherKubernetesEngineConfig{
+			Network: &mgmt.NetworkConfig{Plugin: ct.provisioningConfig.CNIs[0]},
+			Version: ct.provisioningConfig.RKE1KubernetesVersions[0],
+		},
+		},
+	}
+
+	clusterTemplateRevision, err := clustertemplates.CreateRkeTemplateRevision(ct.client, templateRevisonConfig, clusterTemplate.ID)
+	require.NoError(ct.T(), err)
+
+	nodeAndRoles := []provisioninginput.NodePools{
+		provisioninginput.AllRolesNodePool,
+	}
+
+	log.Info("Create a downstream cluster as the standard user.")
+
+	log.Info("Create a standard user.")
+	user, err := users.CreateUserWithRole(ct.client, users.UserConfig(), "user")
+	require.NoError(ct.T(), err)
+	standardClient, err := ct.client.AsUser(user)
+	require.NoError(ct.T(), err)
+
+	rke1Provider := provisioning.CreateRKE1Provider(ct.provisioningConfig.Providers[0])
+	nodeTemplate, err := rke1Provider.NodeTemplateFunc(standardClient)
+	require.NoError(ct.T(), err)
+
+	_, err = provisioning.CreateProvisioningRKE1ClusterWithClusterTemplate(standardClient, clusterTemplate.ID, clusterTemplateRevision.ID,
+		nodeAndRoles, nodeTemplate, nil)
+	require.Error(ct.T(), err, "User is not expected to be able to create a cluster.")
+	require.Contains(ct.T(), err.Error(), "The clusterTemplateRevision is not found")
```

Comment:
these should be split into 2 tests; the negative use case and the positive one.

Reply:
since this is rbac, I can do it in the next PR as this is mainly focussing on the provisioning part of it.